src/Raven.Client/Http/NodeSelector.cs

```diff
@@ -28,6 +29,43 @@ public NodeSelectorState(Topology topology)
                 UnlikelyEveryoneFaultedChoiceIndex = 0;
             }
 
+            public NodeSelectorState(Topology topology, NodeSelectorState prevState)
+            {
+                Topology = topology;
+                Nodes = topology.Nodes;
+                Failures = new int[topology.Nodes.Count];
+                FastestRecords = new int[topology.Nodes.Count];
+                UnlikelyEveryoneFaultedChoiceIndex = 0;
+
+                var fastestNode = prevState.Nodes.ElementAt(Fastest);
```

Comment:
`Fastest` is *always* `0` - did you mean, `prevState.Fastest` ?

Reply:
Yes I did